### PR TITLE
Added dogecoin currency conversions

### PR DESCRIPTION
**Pull request by JOPHESTUS**
@hawkfalcon might like this :)

### CODE REVIEW

File: src/main/java/com/dsh105/nexus/command/module/general/Doge.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public boolean onCommand(CommandPerformEvent event) {
+        if (event.getArgs().length == 0) {
+            return false;
+        } else {
+            try {
+                HttpResponse<JsonNode> jsonResponse = Unirest.get("https://www.dogeapi.com/wow/v2/")
+                        .field("a", "get_info")
+                        .header("accept", "application/json")
+                        .asJson();
+                                double dogeyer = jsonResponse.getBody().getObject().getJSONObject("data").getJSONObject("info").getDouble("doge_usd");
```
Comment (lol768) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: extraneous spacing on this line.

Also, please rename the variable to something appropriate :P

File: src/main/java/com/dsh105/nexus/command/module/general/Doge.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.json.JSONObject;
+import org.pircbotx.Colors;
+
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.text.ParsePosition;
+
+@Command(command = "doge", needsChannel = false, help = "Doge currency converter",
```
Comment (lol768) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: Might want to explain this is a dogecoin thing in the helptext

File: src/main/java/com/dsh105/nexus/command/module/general/Doge.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+                amtInDogetoUSD = (double) Math.round(amtInDogetoUSD * 100) / 100;
+                amtInUSDtoDoge = (double) Math.round(amtInUSDtoDoge * 100) / 100;
+
+                event.respond(Colors.BOLD + "USD --> Doge: " + Colors.NORMAL + Colors.UNDERLINE + amtInUSDtoDoge);
+                event.respond(Colors.BOLD + "Doge --> USD: " + Colors.NORMAL + Colors.UNDERLINE + amtInDogetoUSD);
+
+
+            } catch (UnirestException e) {
+                e.printStackTrace();
```
Comment (lol768) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: Please send the user an error :heart:

Comment (DSH105) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: Throw another exception so the Gist maker can catch it <3

**Comment by lol768** [comment]
Exchange rate caching might be something to look into in the future. Would be great if we could make an issue for that

**Comment by stuntguy3000** [comment]
Exchange rates almost constantly change. 

EDIT: Unfortunately, I didn't appreciate how right lol768 was and so I now understand the value of caching.

File: src/main/java/com/dsh105/nexus/command/module/general/Doge.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+                amtInDogetoUSD = (double) Math.round(amtInDogetoUSD * 100) / 100;
+                amtInUSDtoDoge = (double) Math.round(amtInUSDtoDoge * 100) / 100;
+
+                event.respond(Colors.BOLD + "USD --> Doge: " + Colors.NORMAL + Colors.UNDERLINE + amtInUSDtoDoge);
+                event.respond(Colors.BOLD + "Doge --> USD: " + Colors.NORMAL + Colors.UNDERLINE + amtInDogetoUSD);
+
+
+            } catch (UnirestException e) {
+                throw new DogeCoinException("message", e);
```
Comment (DSH105) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: _facepalm_

Comment (JOPHESTUS) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: y u facepalm

Comment (DSH105) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: "message" -> very helpful

Comment (lol768) [inline src/main/java/com/dsh105/nexus/command/module/general/Doge.java]: @JOPHESTUS: `Message` is an inappropriate exception message :angry: :rage: :cry:

**Comment by DSH105** [comment]
\o/

**Comment by lol768** [comment]
Travis builds are failing due to the testing issue.

**Pull request merged**